app/src/main/java/org/p2p/wallet/jupiter/ui/settings/presenter/JupiterSwapSettingsPresenter.kt

```diff
@@ -180,16 +181,22 @@ class JupiterSwapSettingsPresenter(
                 loadingMapper.mapLoadingList()
             }
             is SwapState.LoadingTransaction -> {
+                val solToken = tokens.firstOrNull { it.isSol() }
+                val solTokenRate = solToken?.let { swapTokensRepository.getTokenRate(it) }
```

Comment:
loading each time? 
it caching, like was in mapper
```
val tokenRate = swapStateManager.getTokenRate(solToken)
            .filterIsInstance<SwapRateLoaderState.Loaded>()
            .firstOrNull()
            ?.rate
```

Reply:
getTokenRate works only with SwapTokenModel, not with JupiterSwapToken
i'll check what i can do

Reply:
checked
nothing really can be done here
or cache prices in the repository or rewrite RateManager
i can suggest only to skip this rate request and return to this issue if it becomes one :)

Reply:
you can just create  SwapTokenModel.JupiterToken(JupiterSwapToken)

Reply:
or load tokens with org.p2p.wallet.swap.jupiter.interactor.SwapTokensInteractor#getAllTokens